src/JavaCodeGenerator.cs

```diff
@@ -1814,7 +1823,21 @@ private static void AddRegularMethodOverloads(JavaType typeBlock, Method method,
             // --------------
             // Async Maybe<T>
             // --------------
-            string asyncMethodReturnType = (methodReturnType.Body == null ? "Completable" : $"Maybe<{ResponseServiceResponseGenericParameterString(methodReturnType, settings)}>");
+
+            string asyncMethodReturnType;
+            if (methodReturnType.Body != null)
+            {
+                asyncMethodReturnType = $"Maybe<{ResponseServiceResponseGenericParameterString(methodReturnType, settings)}>";
+            }
+            else if (isFluentDelete)
+            {
+                asyncMethodReturnType = "Maybe<Void>";
+            }
+            else
+            {
+                asyncMethodReturnType = "Completable";
```

Comment:
Remind me whey Maybe<Void> is different than Completable?

Reply:
`Maybe<T>` calls one of onSuccess(T), onComplete(), or onError(Throwable) when subscribed, while Completable only has onComplete and onError(Throwable).

Some ARM delete calls return values, so they're represented with `Maybe<TModel>`. In order to have all the standard ARM delete calls conform to InnerSupportsDelete, the calls which *don't* return values have to use the return type `Maybe<Void>` and implement `InnerSupportsDelete<Void>` instead returning `Completable`. Since the argument to onSuccess must be non-null, a `Maybe<Void>` will only ever call onComplete and onError in practice.

Reply:
Ah, right. Sorry you have to keep repeating that to me.